sim/dex.js

```diff
@@ -458,10 +460,8 @@ class ModdedDex {
 		moveCopy.hit = 0;
 		return moveCopy;
 	}
+
 	/**
-	 * While this function can technically return any kind of effect at
-	 * all, that's not a feature TypeScript needs to know about.
-	 *
```

Comment:
Why is this removed? It's still correct.

Reply:
Oh, I see, it's moved to the inner function. If you must have an inner function, I think the comment still belongs here.

Reply:
I left this on `getEffectInner`, but didn't copy the comment to `getEffect`. Intuitively it makes sense on `getEffect` because thats the 'public' method people should call, but at the same time, its not the method using '`// @ts-ignore`' all over the place, so I thought it made more sense to leave it with `getEffectInner`. I'm happy to place it wherever you think appropriate.

(I'm curious as to why `getEffect` doesn't just return `Effect` instead of `PureEffect`?)

Reply:
"That's not a feature TypeScript needs to know about" is the hint here. It returns `PureEffect` because most calls to `getEffect` expect (and are guaranteed) a `PureEffect`, so returning `PureEffect` avoids a lot of casting on the caller side.

Reply:
(And because we're using janky TS-in-JS instead of real TypeScript, casting is really hard.)